docs/BEST_PRACTICES.md

```diff
@@ -1,3 +1,69 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [Order creation best practices](#order-creation-best-practices)
+    - [AFTER a successful payment](#after-a-successful-payment)
+      - [Limitations & Solutions](#limitations--solutions)
+    - [BEFORE a successful payment](#before-a-successful-payment)
+      - [Limitations](#limitations)
+      - [Conclusion](#conclusion)
+- [Payment related best practices](#payment-related-best-practices)
+- [Order/Cart related best practices](#ordercart-related-best-practices)
+- [Deployment best practices](#deployment-best-practices)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Order creation best practices
+#### AFTER a successful payment
+##### Limitations & Solutions
+- It is possible that shop was not reachable for network issues. In that case there will be a successful payment but no order.
+
+  ###### Possible solution:
+  Create an order based on transaction state changes (commercetools messages or subscriptions)
```

Comment:
There are 2 cases where using commercetools messages or subscriptions is recommended. Would it be possible to cover this subject in the additional doc? 

I feel like it won't be that obvious for developers and they will just omit these 2 best practices or only the minority will do it properly and it will be very time-demanding for them.

I totally understand it might be too much for you. However, I think it will increase developer experience and nothing will prevent developers to apply every best practice then.

Reply:
Please check if it addresses your request: https://github.com/commercetools/commercetools-adyen-integration/pull/670#discussion_r622505085

Reply:
Yes, now it gives me the context. Thanks!